### PR TITLE
Port to 1.13

### PR DESCRIPTION
**Pull request by williewillus**
I guess I'll put this out here. WIP until Forge is stable.

If Azanor doesn't explicitly hand off Baubles to someone else, I may make releases of this branch once it's ready, since I need Baubles to port Botania.

Notable changes:
* Cleanups everywhere
* Remove support for extending IBauble directly, you must use caps now.

### CODE REVIEW

**Comment by KingLemming** [comment]
Lemme confirm with him, but I was actually looking at doing some rewrites for this before going to 1.13.

**Comment by KingLemming** [comment]
Also I'd really recommend that Botania adopt a soft-dep approach to Baubles. There shouldn't be a need for a hard-dep.

**Comment by williewillus** [comment]
This should be mostly ready now. Cleaned up a lot of stuff to reduce global state.

**Comment by williewillus** [comment]
any update on this?

**Comment by KingLemming** [comment]
I've got permission to publish, I just have to go over it and make sure it works.

Is Botania soft-dep now?

**Comment by williewillus** [comment]
no, it's too tightly woven in and I haven't extricated everything.
It's probably easer to debug that process anyways if I have it compiling and running first

**Comment by williewillus** [comment]
though do note when testing that https://github.com/MinecraftForge/MinecraftForge/issues/5500 is still a problem so interdimensional teleports will lose caps until that's fixed

**Comment by itsmeow** [comment]
Any news on this + a 1.14 version? I also need an updated baubles for a few minor features in the 1.14 version of Better Animals Plus. iirc 1.14 didn't change too much that wasn't entity AI and mappings from 1.13 so it really should just be intellij -> migrate plus a few fixes

**Comment by williewillus** [comment]
I don't have time for this anymore so someone can take this branch and continue working on it from there

**Comment by itsmeow** [comment]
Alright, I'm gonna make a workspace and see what I can get done with this on 1.14.

**Comment by itsmeow** [comment]
Okay, so I did the majority of the tedious porting work and got all the mappings and compilation issues sorted, as well as updated the container to the new types system, however there's quite a few issues that make it basically unusable, including syncing issues (I never got the tracking thing working right, will have to look at that) and there's an index out of bounds from vanilla's attempts to sync the container.

Basically- I'm outside of my abilities and I need someone to step in and take this from me
https://github.com/itsmeow/Baubles

**Comment by KingLemming** [comment]
I'm waiting on Forge to get a couple of things ready with 1.14.4...I'm staying aprised of the situation though.

**Comment by raoulvdberge** [comment]
Any updates on this one?

**Comment by Wychmire** [comment]
Now that Forge is on 1.16.1, has there been any progress here?

**Comment by williewillus** [comment]
Closing this since I no longer need it.

Baubles should be considered permanently dead. Curios (https://github.com/TheIllusiveC4/Curios) is a replacement on 1.13+ with much nicer API and implementation. Botania has moved to using it with no issues at all.
